bootstring/src/main/scala/cats/bootstring/Base.scala

```diff
@@ -56,62 +98,79 @@ object Base {
+      if (int < arr.size) {
+        arr(int)
+      } else {
+        throw new RuntimeException(s"There is no digit in this base which corresponds to $int.")
+      }
 
-  def from(
-      baseValue: Int,
-      codePointList: List[Char],
-      codePointToDigitF: Int => Int
-  ): Either[String, Base] =
-    if (codePointList.size === baseValue) {
-      val codePointArray: Array[Char] = codePointList.toArray
-      Right(
-        new Base {
-          override val value: Int = baseValue
-
-          final override def unsafeDigitToCodePoint(digit: Int, uppercase: Boolean): Int =
-            if (uppercase) {
-              codePointArray(digit).toUpper.toInt
-            } else {
-              codePointArray(digit).toLower.toInt
-            }
-
-          final override def unsafeCodePointToDigit(codePoint: Int): Int =
-            codePointToDigitF(codePoint)
-
-          final override def digitToCodePoint(
-              digit: Int,
-              uppercase: Boolean): Either[String, Int] =
-            ApplicativeError[Either[Throwable, *], Throwable]
-              .catchNonFatal(
-                unsafeDigitToCodePoint(digit, uppercase)
-              )
-              .leftMap(_.getLocalizedMessage)
-
-          final override def codePointToDigit(codePoint: Int): Either[String, Int] =
-            ApplicativeError[Either[Throwable, *], Throwable]
-              .catchNonFatal(
-                unsafeCodePointToDigit(codePoint)
-              )
-              .leftMap(_.getLocalizedMessage)
-        }
-      )
-    } else {
-      Left(
-        s"Base values must be equal to the size of the defined code points for each base digit. codePointList size is ${codePointList.size}, base value is ${baseValue}")
-    }
+    unsafeFrom(
+      36,
+      d => intToCodePointDigit(d, lowercaseArray),
+      d => intToCodePointDigit(d, uppercaseArray),
+      codePointDigitToInt)
+  }
 
+  /**
+   * Create a [[Base]] value.
+   *
+   * @param baseValue
+   *   The number of digits in the base.
+   * @param intToLowerCodePointF
+   *   A function to convert an integer value to a the Unicode code point for the digit with the
+   *   corresponding value in this base with a lower case representation. This may throw on
+   *   invalid input.
+   * @param intToUpperCodePointF
+   *   A function to convert an integer value to a the Unicode code point for the digit with the
+   *   corresponding value in this base with an upper case representation. This may throw on
+   *   invalid input.
+   * @param codePointDigitToIntF
+   *   A function to convert a Unicode code point representing a digit in this base to the
+   *   corresponding numerical value. This may throw on invalid input.
+   */
   def unsafeFrom(
       baseValue: Int,
-      codePointList: List[Char],
-      codePointToDigit: Int => Int
+      intToLowerCodePointF: Int => Int,
+      intToUpperCodePointF: Int => Int,
+      codePointDigitToIntF: Int => Int
```

Comment:
Note that on Scala 3 there is no specialization, so using these will involve boxing. Not sure if that's a big deal.

Reply:
Yeah...that's probably going to be an issue. This is performance critical code. I'll update that.

Reply:
In that case wouldn't some form of subclassing work better here? Even with specialization 3 lambdas is 3 indirections, when they could all be defined in the same concrete class.

Reply:
Yeah, you're absolutely correct. I'll update it to that encoding and I'll probably unseal the class too. I doubt anyone else will ever make an instance of this, but in the event they do I want the encoding to not prevent performant implementations.

If I unseal it, I'll probably drop `unsafeFrom`.